nautobot/core/views/renderers.py

```diff
@@ -0,0 +1,222 @@
+                if request.POST.get("_all"):
+                    if view.filterset_class is not None:
+                        pk_list = [obj.pk for obj in self.filterset_class(request.POST, model.objects.only("pk")).qs]
+                    else:
+                        pk_list = model.objects.values_list("pk", flat=True)
+                else:
+                    pk_list = request.POST.getlist("pk")
+                    initial = {
+                        "pk": pk_list,
+                        "return_url": view.get_return_url(request),
+                    }
+                    form = form_class(initial=initial)
+                table = self.construct_table(view, pk_list=pk_list)
+                return_url = view.get_return_url(request)
+            elif view.action == "bulk_create":
+                form = view.get_form()
+                return_url = view.get_return_url(request)
+                if request.data:
+                    table = data.get("table")
+            elif view.action == "bulk_update":
+                if request.POST.get("_all") and view.filterset_class is not None:
+                    pk_list = [obj.pk for obj in view.filterset_class(request.POST, view.queryset.only("pk")).qs]
+                else:
+                    pk_list = request.POST.getlist("pk")
+                if "_apply" in request.POST:
+                    form = form_class(model, request.POST)
+                    restrict_form_fields(form, request.user)
+                else:
+                    # Include the PK list as initial data for the form
+                    initial_data = {"pk": pk_list}
+                    # Check for other contextual data needed for the form. We avoid passing all of request.GET because the
+                    # filter values will conflict with the bulk edit form fields.
+                    # TODO: Find a better way to accomplish this
+                    if "device" in request.GET:
+                        initial_data["device"] = request.GET.get("device")
+                    elif "device_type" in request.GET:
+                        initial_data["device_type"] = request.GET.get("device_type")
+
+                    form = form_class(model, initial=initial_data)
+                    restrict_form_fields(form, request.user)
+                table = self.construct_table(view, pk_list=pk_list)
+                return_url = view.get_return_url(request)
+        context = {
+            "action_buttons": valid_actions,
+            "active_tab": "csv-data",
+            "changelog_url": changelog_url,
+            "content_type": content_type,
+            "editing": instance.present_in_database,
+            "fields": form_class(model).fields if form_class else None,
+            "filter_form": self.get_filter_form(view, request),
+            "form": form,
+            # I am keeping "object" and "obj" to keep the template changes needed minimally invasive.  # "object" is used in object_detail.html template.
```

Comment:
My concern here is whether this will result in us being locked into this pattern "forever". As soon as we have plugins using the new templates, we won't be able to remove one of these without it being a breaking change.

Reply:
After going back and forth I think as long as we have plugins providing extra context (instead of their own), aligning on the eventual context key for `object`, this will cause the least amount of headache in this transition.